json.c

```diff
@@ -16,6 +16,30 @@
 
 #include "drm_info.h"
 
+// HDR definitions copied from linux/include/uapi/drm/drm_mode.h
```

Comment:
We can `#include <libdrm/drm_mode.h>` to get these. Meson can ensure we have recent enough libdrm headers (and download a recent copy if the system libdrm is too old).

Reply:
These came from the Linux uapi header. I don't see hdr_output_metadata in upstream libdrm yet.

I could send another change to libdrm, and I could additional guard the local definitions with an #ifdef. Is this the right way to check in Meson?

if cc.has_header_symbol('drm_mode.h', 'hdr_output_metadata', dependencies: [libdrm])
  add_project_arguments('-DHAVE_HDR_OUTPUT_METADATA', language: 'c')
endif

Reply:
>I don't see hdr_output_metadata in upstream libdrm yet.

Ah, I see. Yeah, it would be nice to sync libdrm headers with upstream.

I'm a bit surprised `hdr_metadata_infoframe` isn't prefixed with "drm_".

>Is this the right way to check in Meson?

Yeah, sounds like this would do it.

Reply:
Done.

I also tried adding the headers to libdrm, but found that they're synced automatically from drm-next's uapi headers with "make headers_install". The kernel definitions also need some shuffling, as discussed with vsyrjala on #dri-devel.